src/coord/Axis.ts

```diff
@@ -129,7 +129,12 @@ class Axis {
             fixExtentWithBands(extent, (scale as OrdinalScale).count());
         }
 
-        return linearMap(data, NORMALIZED_EXTENT, extent, clamp);
+        if ((extent[0] === extent[1]) && (data < 0 || data > 1)) {
+            return data * extent[0] * 2;
```

Comment:
I still think the equation used here is too tricky. It heavily depends on what the data is and can't be explained when data is not -0.5 or 1.5.

Reply:
data here is promised to be -0.5, 0.5 or 1.5 in scale.normalize() when extent[0] === extent[1]

Reply:
@pissang Hi a new issue https://github.com/apache/echarts/issues/17014 opened 2 days ago is also relavant. I think such issues can occur quite often during use of Echarts. We shall solve this problem through this PR soon.

Reply:
Yeah this bug should be fixed. But as I mentioned in the above comment. The fomular used here is just too magical and I can't be convinced it is meaningful on all cases.  For example: 
+ Why choose mutpliying 2, not 3 or any other numbers. 
+ Here we check data < 0 || data > 0 and assume `normalize` will always return -0.5, 1.5.  But what if we tweaked the result of `normalize` and return other values instead? Will we forget to tweak the fomular to fit the expect result? It's a kind of leaky abstraction

Reply:
If I'm convinced. Then comments should be added to explain why this fomular will work

Reply:
After consider over this issue, I think the main problem is still in `linearMap`. I changed it and let it return NaN under permission when `val` has exceeded `domain`. And for now the permission is by default `false` and I let it be true at Axis.ts only. After testing in the future we can open permissions at other places.
Also please check the CI test because it keeps failing at collecting changed files.